blog/_posts/2018-05-01-extensible-broadcast-fusion.md

```diff
@@ -0,0 +1,62 @@
++---
+ +layout: post
+ +title:  "Extensible broadcast fusion"
+ +author: <a href="https://github.com/mbauman">Matt Bauman (Julia Computing)</a>
+ +---
+
+Julia version 0.7 brings with it an exciting new feature: the ability to customize broadcast fusion! This recently-merged change is the culmination of a long iterative design process that involved many members of the community. We have converged on a highly extensible interface that should satisfy many use-cases.  In this blog post I'll explain why this is a big deal by reviewing some of the key features and just scratch the surface of all that is possible with this new design. I'm quite certain that our enterprising community will come up with many more clever ways to exploit this new structure in the future.
+
+Julia now uses a first-class datastructure to "lazily" represent a fused broadcast expression before executing it. If you're not a package developer this may not mean all that much to you, but you'll still reap the many rewards. In Base Julia and its standard libraries alone, this means:
+
+* Ranges can now identify cases where they can compute the result in terms of a new range. For example, the expression `((1:10000) .+ 20) .* 7` doesn't need to allocate a vector for 10,000 elements — it doesn't even need to do 10,000 computations. It can simply operate in terms of the start, stop, and step to compute the result: `147:7:70140`.
```

Comment:
I'd put `Range`s in backticks like so, similarly for `BitArray`s and perhaps `LinearAlgebra` below.